pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/vsphere_test.go

```diff
@@ -94,6 +94,31 @@ var _ = Describe("VSphere Provider Config", Label("vSphereProviderConfig"), func
 		})
 	})
 
+	Context("StaticIP", func() {
+
+		BeforeEach(func() {
+			machineProviderConfig := machinev1beta1resourcebuilder.VSphereProviderSpec().
+				WithZone(usCentral1a).WithIPPool().
+				Build()
+
+			providerConfig = VSphereProviderConfig{
+				providerConfig: *machineProviderConfig,
+				infrastructure: configv1resourcebuilder.Infrastructure().AsVSphereWithFailureDomains("vsphere-test", nil).Build(),
+			}
+		})
+
+		It("contains an AddressesFromPools block", func() {
+			Expect(providerConfig.providerConfig.Network.Devices[0].AddressesFromPools).To(Not(BeEmpty()))
+		})
+
+		It("returns networking with AddressesFromPools and the configured network name from failure domain", func() {
+			expected, err := providerConfig.InjectFailureDomain(providerConfig.ExtractFailureDomain())
+			Expect(err).To(BeNil())
```

Comment:
Here we normally prefer to use:
```suggestion
			Expect(err).To(Not(HaveOccurred()))
```

As that provides a nicer output:
```
    Unexpected error:
        <*errors.errorString | 0xc00002c5d0>:
        this is a very bad error
        {
            s: "this is a very bad error",
        }
    occurred
```
as opposed to:
```
    Expected
        <*errors.errorString | 0xc0000a0590>:
        this is a very bad error
        {
            s: "this is a very bad error",
        }
    to be nil
```

Reply:
Makes sense.  I'll update this as well.